File: t6/Rectangle.cpp
/**
 * Class that defines an interface of a rectangle.
 *
 * @author João Bolsson (jvmarques@inf.ufsm.br)
 * @since 2017, 17 May.
 */

#ifndef RECTANGLE
#define RECTANGLE

#include "Shape.cpp"

class Rectangle : public Shape {

private:
    int width, height;

public:

    /**
     * Default construct.
     *
     * @param width Rectangle width.
     * @param height Rectangle height.
     */
    Rectangle(int width, int height) {
        this->width = width;
        this->height = height;
    }

    /**
     * @return The area of this rectangle.
     */
    double getArea() {
        return width * height;
    }
};

#endif

File: t6-java/src/main/Rectangle.java
package main;

/**
 * Class that defines a rectangle.
 *
 * @author João Bolsson (joaovictorbolsson@gmail.com)
 * @since 2017, 27 May.
 */
public class Rectangle extends Shape {

    private final int height, width;

    /**
     * Default construct.
     *
     * @param width Rectangle width.
     * @param height Rectangle height.
     */
    public Rectangle(final int width, final int height) {
        this.height = height;
        this.width = width;
    }

    /**
     * @return Rectangle area.
     */
    @Override
    public double getArea() {
        return height * width;
    }

    @Override
    public String toString() {
        return "Retângulo";
    }

}
